packages/cli/src/dev.ts

```diff
@@ -184,44 +184,50 @@ export default async function dev(projectDir: string) {
-// as you would import it if it was built with preconstruct build
-// this file is slightly different to some others though
-// it has a require hook which compiles your code with Babel
-// this means that you don't have to set up @babel/register or anything like that
-// but you can still require this module and it'll be compiled
-
-// this bit of code imports the require hook and registers it
-let unregister = require(${JSON.stringify(
-                normalizePath(
-                  path.relative(
-                    distDirectory,
-                    path.dirname(require.resolve("@preconstruct/hook"))
-                  )
-                )
-              )}).___internalHook(typeof __dirname === 'undefined' ? undefined : __dirname, ${JSON.stringify(
-                normalizePath(path.relative(distDirectory, project.directory))
-              )}, ${JSON.stringify(
-                normalizePath(path.relative(distDirectory, pkg.directory))
-              )});
+          ];
 
-// this re-exports the source file
-module.exports = require(${JSON.stringify(
-                normalizePath(path.relative(distDirectory, entrypoint.source))
-              )});
+          if (entrypoint.json.type !== "module") {
+            promises.push(
+              fs.writeFile(
+                path.join(
+                  entrypoint.directory,
+                  validFieldsForEntrypoint.main(entrypoint)
+                ),
+                `"use strict";
+  // this file might look strange and you might be wondering what it's for
+  // it's lets you import your source files by importing this entrypoint
+  // as you would import it if it was built with preconstruct build
+  // this file is slightly different to some others though
+  // it has a require hook which compiles your code with Babel
+  // this means that you don't have to set up @babel/register or anything like that
+  // but you can still require this module and it'll be compiled
+  
+  // this bit of code imports the require hook and registers it
+  let unregister = require(${JSON.stringify(
+    normalizePath(
+      path.relative(
+        distDirectory,
+        path.dirname(require.resolve("@preconstruct/hook"))
+      )
+    )
+  )}).___internalHook(typeof __dirname === 'undefined' ? undefined : __dirname, ${JSON.stringify(
+                  normalizePath(path.relative(distDirectory, project.directory))
+                )}, ${JSON.stringify(
+                  normalizePath(path.relative(distDirectory, pkg.directory))
+                )});
+  
+  // this re-exports the source file
+  module.exports = require(${JSON.stringify(
+    normalizePath(path.relative(distDirectory, entrypoint.source))
+  )});
+  
+  unregister();
+  `
+              )
+            );
+          }
 
-unregister();
-`
-            ),
-          ];
-          if (entrypoint.json.module) {
+          if (entrypoint.json.module || entrypoint.json.type === "module") {
```

Comment:
general question - should we even bother supporting "mixed" packages where some entrypoints are of type module and some are of type commonjs? IMHO the added complexity isn't quite worth it